### ISSUE TITLE
All getters get called without being accessed (even before data is loaded)

### ISSUE THREAD

**Issue by Sarke** [FiguredLimited/vue-mc#34]
I have an issue with all the getters being hit, even before the data is loaded, which gives a bunch of nasty undefined.

I have created this to show the issue:

https://runkit.com/sarke/5a5e956c9fbc9a0012585e0f

Getters should only be called when directly accessed.

**Comment by Sarke** [comment]
It appears that https://github.com/FiguredLimited/vue-mc/blob/37ed8a07f76ad1951671bba5d53816c6b0744ba3/src/utils.js#L20 is to blame.  They `typeof obj[name]` statements calls it.

A way around this is to add the following above that line:

```javascript
if (typeof Object.getOwnPropertyDescriptor(obj, name).get !== 'undefined') {
	continue;
}
```

**Comment by rtheunissen** [comment]
Such a good find, thank you @Sarke ✨

**Comment by rtheunissen** [comment]
`set` as well, right? Implementing this now.

**Comment by Sarke** [comment]
> Such a good find, thank you @Sarke ✨
No problem, thanks for fixing it!

> `set` as well, right?
No, I don't believe the setter function gets called because there is no value assignment.

**Comment by rtheunissen** [comment]
> No, I don't believe the setter function gets called because there is no value assignment.

:man_facepalming:

**Comment by rtheunissen** [comment]
This has been fixed and published under 0.2.4
